### PR TITLE
Feature/466

### PR DESCRIPTION
**Pull request by shaneknows**
Adding parsing of commands for populating search filters. In the future this can be passed to the search results page to populate the filters, but for now just populating them on the search page

### CODE REVIEW

**Review by lwvantassel — DISMISSED** [review]

File: lib/models/SearchCriteria.dart [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  // Constructor to parse params
+  SearchCriteria.withList(List<String> params) {
+    DateTime? _potentialStart;
+    DateTime? _potentialEnd;
+    String _potentialKeyword = "";
+
+    if (params.isEmpty) {
+      // default init
+      _potentialStart = DateTime.now();
+      _potentialEnd = DateTime.now();
```
Comment (umgcjack) [inline lib/models/SearchCriteria.dart]: There is a null check, and fallback to `DateTime.now()`, when you assign to `this.startDate` and `this.endDate`.
You can remove this part of the `if` statement and make it `if (params.isNotEmpty) ...` to simplify this.

File: lib/ui/search.dart [hunk 2]
```diff
@@ -5,10 +5,14 @@ import 'package:intl/intl.dart';
 import 'package:syncfusion_flutter_datepicker/datepicker.dart';
 import 'package:summer2022/ui/bottom_app_bar.dart';
 
+import '../models/SearchCriteria.dart';
+
 class SearchWidget extends StatefulWidget {
+  final List<String> parameters;
+  static const defaultValue = <String>[];
+  const SearchWidget({this.parameters = defaultValue});
```
Comment (umgcjack) [inline lib/ui/search.dart]: You don't use `defaultValue` anywhere else, so you could do `this.parameters = const []` if you wanted.

File: lib/ui/search.dart [hunk 3]
```diff
@@ -26,10 +30,25 @@ class SearchWidgetState extends State<SearchWidget> {
   final DateFormat _dateFormat = DateFormat("M/d/yyyy");
   DateTime _start = DateTime.now();
   DateTime _end = DateTime.now();
+  String _keyword = "";
+  TextEditingController keywordInput = TextEditingController();
+
+  // Apply and passed in search parameters to the filters
+  void applyFilters() {
+    if (this.widget.parameters.isEmpty) return;
+    SearchCriteria filters = SearchCriteria.withList(this.widget.parameters);
```
Comment (umgcjack) [inline lib/ui/search.dart]: Canonical Dart prefers using `final` over explicit typing.
See: https://dart.dev/guides/language/effective-dart/usage#do-follow-a-consistent-rule-for-var-and-final-on-local-variables

**Review by umgcjack — DISMISSED** [review]
Nice work!

**Review by mdconatser — APPROVED** [review]

**Review by lwvantassel — APPROVED** [review]

**Review by shaneknows — COMMENTED** [review]
Reviewed and applied

**Pull request merged**